spring-jdbc/src/main/java/org/springframework/jdbc/datasource/init/ScriptUtils.java

```diff
@@ -270,27 +269,19 @@ public static void executeSqlScript(Connection connection, EncodedResource resou
 								warningToLog = warningToLog.getNextWarning();
 							}
 						}
-					}
-					catch (SQLException ex) {
+					} catch (SQLException ex) {
 						boolean dropStatement = StringUtils.startsWithIgnoreCase(statement.trim(), "drop");
 						if (continueOnError || (dropStatement && ignoreFailedDrops)) {
 							if (logger.isDebugEnabled()) {
 								logger.debug(ScriptStatementFailedException.buildErrorMessage(statement, stmtNumber, resource), ex);
 							}
-						}
-						else {
+						} else {
 							throw new ScriptStatementFailedException(statement, stmtNumber, resource, ex);
 						}
 					}
 				}
-			}
```

Comment:
Please revert. I think this will catch all exceptions, not just when trying to close.